### PR TITLE
support asynchronous tick data fetching

### PR DESCRIPTION
**Pull request by moodysalem**

### CODE REVIEW

**Review by willpote — COMMENTED** [review]
Does the front-end use these functions? If so does it need to be updated to handle promises?

File: src/entities/pool.ts [hunk 1]
```diff
@@ -20,14 +20,65 @@ interface StepComputations {
...
+  async nextInitializedTickWithinOneWord(
+    _tick: number,
+    _zeroForOne: boolean,
+    _tickSpacing: number
+  ): Promise<[number, boolean]> {
+    throw new Error('No tick data provider was given')
+  }
+}
+
+const NO_TICK_DATA_PROVIDER_DEFAULT = new NoTickDataProvider()
```
[the remark this answers is not in the document]
Comment (moodysalem) [reply src/entities/pool.ts:69]: empty array would be a valid list of tick data

**Comment by moodysalem** [comment]
> Does the front-end use these functions? If so does it need to be updated to handle promises?

it doesnt use them yet in the v3 sdk, but yeah it will have to be made asynchronous which is pretty easy

**Review by willpote — APPROVED** [review]

**Review by Chucky-ux — COMMENTED** [review]
Looks good

**Pull request merged**